### DISCUSSION TITLE
GraphQL Interface without fields leads to error in mutation

### DISCUSSION THREAD

**Discussion by lfr-ableton** [community/community#24803] [Apps, API and Webhooks]
<p>Hi,</p>
<p>when using the mutation <a href="https://developer.github.com/v4/mutation/addlabelstolabelable/" title="addLabelsToLabelable" rel="nofollow noopener">addLabelsToLabelable</a> it returns two fields</p>
<ul>
<li>
<strong>clientMutationId</strong> (that one is always null in my case)</li>
<li><strong>labelable</strong></li>
</ul>
<p>I would like to use <strong>labelable</strong> since I label a <a href="https://developer.github.com/v4/object/pullrequest/" rel="nofollow noopener">PullRequest</a>, which implement the <a href="https://developer.github.com/v4/interface/labelable/" rel="nofollow noopener">Labelable</a> interface. But that interface has no fields. So I cannot give any field in the mutation, which leads to the error:</p>
<pre><code>Field must have selections (field 'labelable' returns Labelable but has no selections. Did you mean 'labelable { ... }'?)
</code></pre>
<p>So the only “workaround” is to use  <strong>clientMutationId </strong> instead and receive the null. But I would like to receive e.g. the title of the labeled Pull Request. Is that possible?</p>
<p>Thanks and Cheers,</p>
<p>Lars</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>From what I understand, you should be able to use the same <code>... on PullRequest</code> syntax that you would use in a query that returns multiple object types. So something like this:</p>
<pre><code>mutation {
  addLabelsToLabelable(input: {clientMutationId: "foo", labelableId: "foo", labelIds: [1]}) {
    labelable {
      ... on PullRequest {
        title
      }
    }
  }
}
</code></pre>
<p>Let us know if you have more questions.</p>

**Comment by lfr-ableton** [comment]
<p>Thanks a lot,</p>
<p>that does the job!</p>
<p>Cheers,</p>
<p>Lars</p>
